src/java/org/apache/cassandra/schema/TableMetadata.java

```diff
@@ -461,57 +461,125 @@ public static boolean isSuperColumnMapColumnName(ByteBuffer columnName)
         return !columnName.hasRemaining();
     }
 
+    /**
+     * Method that compares two TableMetadata objects. This is a modified version of validateCompatibility that is used
+     * when checking the compatibility between the schema metadata from an SSTable against the schema metadata from a
+     * CQL table.
+     *
+     * The serialization header of the SSTable does not contain exactly the same information available in the schema of
+     * a CQL table, so the comparison needs to be adapted to the information available.
+     *
+     * For example, the serialization header does not contain the partition key columns: it only contains the composite
+     * type of the whole partition key. For this reason, the comparison must be between the partition key types contained
+     * in the two metadata objects, rather than comparing the individual column as validateCompatibility does.
+     * This comparison is sufficient anyway because the composite types are the same only if their components are of the
+     * same type and in the same order.
+     *
```

Comment:
Thanks for adding the Javadoc, it's really helpful. Maybe we should also make it clear that we are only comparing the table names while the keyspace name and the table id are not checked.

Reply:
Good point, I added a paragraph to the comment to clarify this.